src/helics/core/queryHelpers.cpp

```diff
@@ -14,14 +14,16 @@ namespace helics {
 static void addTags(Json::Value& v, const BasicHandleInfo& bhi)
 {
     if (bhi.tagCount() > 0) {
-        v["tags"] = Json::arrayValue;
+        Json::Value tagBlock = Json::objectValue;
         for (size_t ii = 0; ii < bhi.tagCount(); ++ii) {
             const auto& tg = bhi.getTagByIndex(ii);
-            Json::Value tagBlock;
-            tagBlock["name"] = tg.first;
-            tagBlock["value"] = tg.second;
-            v["tags"].append(tagBlock);
+            if (tg.second != "true") {
+                tagBlock[tg.first] = tg.second;
+            } else {
+                tagBlock[tg.first] = true;
```

Comment:
So you think I should just leave everything as strings in the output?

Reply:
I think so -- at least in JS the check to see if the tag is defined is the same whether its a string or boolean, and internally HELICS converts it to a string on load so it doesn't matter.

I'm not sure about consuming the json from other languages, but just telling them it will always be a string seems simpler rather than making a potential corner case that needs to be handled in a different way.